crayon_wp.class.php

```diff
@@ -779,6 +779,15 @@ public static function class_tag($matches) {
         $atts = $matches[5];
         $content = $matches[6];
 
+        // We don't support nested span/pre/code, strip them out and only.
+        // parse the top level. If we don't do this, we'll possibly nest
+        // [crayon] tags and bad things happen.
+        // We could get fancier with the regex to ensure that the tags match,
+        // but because this is raw HTML if the tags don't match it was already
+        // busted anyway.
+        $strip_regex = '#</?(pre|code|span)[^>]*>#i';
```

Comment:
This may not be right place for this, since this method deals with setting the class on the generated element. Do you want to strip out the tags before formatting? This could have false positives if we're displaying HTML as you mentioned, in particular the popular span tag